### PR TITLE
Fix several byte / unicode issues for Python 3

### PR DESCRIPTION
**Pull request by charlesccychen**
This change fixes some compatibility issues with running on Python 3.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by charlesccychen** [comment]
Run Python PostCommit

**Comment by charlesccychen** [comment]
R: @tvalentyn @aaltay @markflyhigh

File: sdks/python/apache_beam/io/avroio.py [hunk 1]
```diff
@@ -226,7 +226,7 @@ def read_meta_data_from_file(f):
     meta = header['meta']
 
     if datafile.CODEC_KEY in meta:
-      codec = meta[datafile.CODEC_KEY]
+      codec = meta[datafile.CODEC_KEY].decode('utf-8')
```
Comment (RobbeSneyders) [inline sdks/python/apache_beam/io/avroio.py]: When running the avroio tests in Python 3, this results in the following error for me:
```
File "/home/robbe/workspace/beam/sdks/python/apache_beam/io/avroio.py", line 355, in _decompress_bytes
    raise ValueError('Unknown codec: %r' % codec)
ValueError: Unknown codec: 'null'
```
I've used another approach and have fixed all but 2 Python 3 errors in avroio. see #7644.

Comment (tvalentyn) [reply sdks/python/apache_beam/io/avroio.py]: Approach in https://github.com/apache/beam/pull/7644 makes sense to me. Once merged - @charlesccychen  PTAL if you still encounter the errors and rebase the PR on top of master. Thank you.

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py [hunk 2]
```diff
@@ -1046,7 +1046,7 @@ def byte_array_to_json_string(raw_bytes):
   @staticmethod
   def json_string_to_byte_array(encoded_string):
     """Implements org.apache.beam.sdk.util.StringUtils.jsonStringToByteArray."""
-    return unquote(encoded_string)
+    return unquote(encoded_string).encode('utf-8')
```
[the remark this answers is not in the document]
Comment (tvalentyn) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: Thanks for your comments @RobbeSneyders. Let's use unquote_to_bytes.

**Pull request closed without merging**